pkg/queue/health/probe.go

```diff
@@ -17,18 +17,80 @@ limitations under the License.
 package health
 
 import (
+	"crypto/tls"
+	"errors"
+	"fmt"
 	"net"
+	"net/http"
 	"time"
+
+	corev1 "k8s.io/api/core/v1"
 )
 
+// HTTPProbeConfigOptions holds the HTTP probe config options
+type HTTPProbeConfigOptions struct {
+	Timeout time.Duration
+	*corev1.HTTPGetAction
+}
+
+// TCPProbeConfigOptions holds the TCP probe config options
+type TCPProbeConfigOptions struct {
+	SocketTimeout time.Duration
+	Address       string
+}
+
 // TCPProbe checks that a TCP socket to the address can be opened.
 // Did not reuse k8s.io/kubernetes/pkg/probe/tcp to not create a dependency
 // on klog.
-func TCPProbe(addr string, socketTimeout time.Duration) error {
-	conn, err := net.DialTimeout("tcp", addr, socketTimeout)
+func TCPProbe(config TCPProbeConfigOptions) error {
+	conn, err := net.DialTimeout("tcp", config.Address, config.SocketTimeout)
 	if err != nil {
 		return err
 	}
 	conn.Close()
 	return nil
 }
+
+// HTTPProbe checks that HTTP connection can be established to the address.
+func HTTPProbe(config HTTPProbeConfigOptions) error {
+	httpClient := &http.Client{
+		Transport: &http.Transport{
+			DisableKeepAlives: true,
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: true,
+			},
+		},
+		Timeout: config.Timeout,
+	}
+	url := fmt.Sprintf("%s://%s:%d%s", config.Scheme, config.Host, config.Port.IntValue(), config.Path)
```

Comment:
Why not construct `url.URL` object and then use its `String` method?

Reply:
If we build url.URL object then would we end up with something like below

```
	url := url.URL{
		Scheme: fmt.Sprintf("%s", config.Scheme),
		Host:   fmt.Sprintf("%s:%d", config.Host, config.Port.IntValue()),
		Path:   config.Path,
	}
```

Not sure if we gain anything about making building URL object.

Reply:
Well:

1. Scheme  can be just assigned (no need to sprintf)
2. Host is just `config.Host + ":" + config.Port.String()` (no need to sprintf)
3. you don't have to deal with validating slashes "/" (you don't currently, so in fact a wrong URL can be built)

Reply:
> Scheme can be just assigned (no need to sprintf)

config.Scheme is not string type so that needs to be converted.

I agree with the rest so I will update

Reply:
`URIScheme` is `string` alias. So just string(config.Scheme) will do.